opentelemetry/src/sdk/trace/span_processor.rs

```diff
@@ -138,15 +155,16 @@ impl SpanProcessor for SimpleSpanProcessor {
     }
 
     fn shutdown(&mut self) -> TraceResult<()> {
-        if let Ok(mut exporter) = self.exporter.lock() {
-            exporter.shutdown();
-            Ok(())
-        } else {
-            Err(TraceError::Other(
-                "When shutting down the SimpleSpanProcessor, the exporter's lock has been poisoned"
-                    .into(),
-            ))
+        if self.sender.send(None).is_ok() {
+            if let Err(err) = self.shutdown.recv() {
```

Comment:
What's the advantage of the shutdown channel? Could we store the `JoinHandle` of the thread and join here? Is it that `JoinHandle::join` takes `self` and so we would have to put it inside an `Option` or something similar?

Reply:
Had join handle originally, could probably go back to that for simplicity

Reply:
Probably doesn't matter. I was just curious 🙂